src/components/marker-chart/Canvas.js

```diff
@@ -265,35 +273,37 @@ class MarkerChartCanvas extends React.PureComponent<Props, State> {
       if (rowIndex > 0 && name === markerTimingRows[rowIndex - 1].name) {
         continue;
       }
-      const displayedName =
-        name.length < MARKER_LABEL_MAX_LENGTH
-          ? name
-          : name.slice(0, MARKER_LABEL_MAX_LENGTH) + '…';
+      const fittedText = textMeasurement.getFittedText(name, MARGIN_LEFT);
       const y = rowIndex * rowHeight - viewportTop;
-      ctx.fillText(displayedName, 5, y + TEXT_OFFSET_TOP);
+      ctx.fillText(fittedText, 5, y + TEXT_OFFSET_TOP);
     }
   }
 
   hitTest = (x: CssPixels, y: CssPixels): IndexIntoMarkerTiming | null => {
+    if (x < MARGIN_LEFT) {
```

Comment:
nit: I think you can use `MARGIN-LEFT - MARKER_DOT_RADIUS` because a dot marker can be rendered at this location, eg this one at the top left: https://deploy-preview-1286--perf-html.netlify.com/public/dfc1b84f9982b4d67e0c1079bad2c582cde1f32f/marker-chart/?globalTrackOrder=0-1-2-3&hiddenGlobalTracks=1-2-3&hiddenLocalTracksByPid=2399-0-1&localTrackOrderByPid=2399-0-1~2422-0~&thread=0&v=3&markerSearch=BHR